contents/javascript/Javascript-framework-Angular.md

```diff
@@ -0,0 +1,231 @@
+    ```
+    In the code above, parameters `x` and `y` are declared to have the type `number`. Thus, the compiler shows an error when the function is called with strings as inputs to the function.
+    
+    Another advantage of TypeScript is _Type Inference_ - TypeScript makes typing a bit easier and less explicit by the usage of type inference. Let us see how.
+    
+    ```typescript
+    let a = "some text";
+    let b = 123;
+    a = b; // Compiler Error: Type 'number' is not assignable to type 'string'
+    ```    
+    
+    In the above code snippet, we are not explicitly defining `a: string` with a type annotation. TypeScript infers the type of the variable based on the value assigned to the variable. The value of `a` is a string and hence the type of `a` is inferred as `string`.
+    Similarly, the type of `b` is inferred as `number`. Therefore, while inferring types, the compiler flags an error saying a `number` type cannot be assigned to a `string` type.
+    
+    You can read this [article](https://medium.com/tech-tajawal/typescript-why-should-one-use-it-a539faa92010) to learn more about the features that TypeScript has to offer.
+    
+* **Angular CLI** - Angular CLI stands for Angular Command Line Interface. It is a command line tool for managing Angular apps. 
+    With the CLI you can perform the following commands to ease the development process.
+     - Generate a new Angular application with - `ng new`
+     - Generate Angular files - `ng generate`
+     - Build your application for deployment - `ng build`
+     
+    You can refer to the [official documentation](https://cli.angular.io/) for a complete list of the CLI commands.
+    
+    The Angular CLI also helps developers follow good development practices. For example, 
+    The Angular CLI has out-of-the-box integration with <tooltip content="Codelyzer is an open source tool that checks whether the pre-defined coding guidelines have been followed">**Codelyzer**</tooltip>.
+    Codelyzer lints your code against the [official Angular style guide](https://angular.io/guide/styleguide). With Angular CLI, you can simply run the command `ng lint` to get an analysis of whether your code follows the "best practices".
+        
+
+This [article](https://medium.freecodecamp.org/best-practices-for-a-clean-and-performant-angular-application-288e7b39eb6f) provides a comprehensive list of best tooling services offered by Angular.
+ 
+### Benefit 2: Angular Allows Cross-platform Development
+
+Using the modern versions of Angular, developers can build applications that live on the web, mobile, or the desktop.
+
+* **Progressive Web Apps** - PWAs are web-based applications that contain most of the benefits of a typical mobile app. The [official documentation](https://developers.google.com/web/progressive-web-apps/) written by Google describes PWAs to be reliable, fast and engaging. With Angular you can develop PWAs and provide an "immersive user experience".
```

Comment:
`"immersive user experience"` not appropriate to use double-quotes here. It's like using air quotes.